File: jvm/core/main/scala/com/webank/eggroll/core/containers/meta/KillContainersResponse.scala
package com.webank.eggroll.core.containers.meta

import com.google.protobuf.ByteString
import com.webank.eggroll.core.meta.Containers

import scala.language.implicitConversions


case class KillContainersResponse(sessionId: String) {
}

object KillContainersResponse {
  implicit def deserialize(byteString: ByteString): KillContainersResponse = {
    val proto = Containers.KillContainersResponse.parseFrom(byteString)
    KillContainersResponse(proto.getSessionId)
  }

  implicit def serialize(src: KillContainersResponse): Array[Byte] = {
    val builder = Containers.KillContainersResponse.newBuilder()
      .setSessionId(src.sessionId)
    builder.build().toByteArray
  }
}

File: jvm/new_core/src/main/java/com/eggroll/core/pojo/KillContainersResponse.java
package com.eggroll.core.pojo;

import com.google.protobuf.ByteString;
import com.webank.eggroll.core.meta.Containers;
import lombok.Data;
import lombok.val;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

@Data
public class KillContainersResponse {

    Logger log = LoggerFactory.getLogger(KillContainersResponse.class);
    private String sessionId;

    public KillContainersResponse deserialize(byte[] bytes){
        KillContainersResponse killContainersResponse = new KillContainersResponse();
        try {
            Containers.KillContainersResponse proto = Containers.KillContainersResponse.parseFrom(bytes);
            killContainersResponse.setSessionId(proto.getSessionId());

        }catch (Exception e){
            log.error("KillContainersResponse.deserialize() error :" ,e);
        }
        return killContainersResponse;
    }

    public byte[] serialize(){
       return Containers.KillContainersResponse.newBuilder()
                .setSessionId(this.sessionId).build().toByteArray();
    }
}
